exalearn-ui/src/app/services/api.service.ts

```diff
@@ -19,8 +19,13 @@ export class ApiService {
 		}
 	}
 
-	getRequest(path: string): Promise<any> {
-		return this.doRequest(path, {
+	getRequest(path: string, params?: any): Promise<any> {
+		let newPath: string = path;
```

Comment:
Also I think you can pass this check, you can just `return this.doRequest(this.parsePath(path), {...` and inside this.parsePath you already have necessary checks for params (also you can check if object is empty, because !!{} = true

Reply:
got it